lib/oauther.ex

```diff
@@ -1,28 +1,46 @@
 defmodule OAuther do
   defmodule Credentials do
+    @type t :: %__MODULE__{
+      consumer_key:    String.t,
+      consumer_secret: String.t,
+      token:           String.t,
+      token_secret:    String.t,
+      method:          :hmac_sha1 | :rsa_sha1 | :plaintext
+    }
+
     defstruct [:consumer_key, :consumer_secret,
                :token, :token_secret, method: :hmac_sha1]
   end
 
+  @type pair_of_strings :: {String.t, String.t}
+  @type header :: {String.t, any}
+  @type headers :: [header]
```

Comment:
I think we don't need `header` and `headers` types, we return header only in one place, in `header/1` function.